consensus/scp/tests/mock_network/scp_network.rs

```diff
@@ -0,0 +1,178 @@
+//! A simulated SCP network.
+
+use crate::mock_network::{
+    scp_node::{SCPNode, SCPNodeSharedData},
+    NodeConfig, TestOptions,
+};
+use mc_common::{
+    logger::{log, Logger},
+    NodeID,
+};
+use mc_consensus_scp::Msg;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::{Arc, Mutex},
+    thread::JoinHandle,
+};
+
+pub struct SCPNetwork {
+    /// NodeID to node.
+    nodes: Arc<Mutex<HashMap<NodeID, SCPNode>>>,
+
+    /// NodeID to the node's thread handle.
+    handles: HashMap<NodeID, JoinHandle<()>>,
+
+    /// NodeID to node name.
+    pub names: HashMap<NodeID, String>,
+
+    /// NodeID to ...???
+    shared_data: HashMap<NodeID, Arc<Mutex<SCPNodeSharedData>>>,
+
+    logger: Logger,
+}
+
+impl SCPNetwork {
+    // Creates a simulated network.
+    pub fn new(node_configs: &[NodeConfig], test_options: &TestOptions, logger: Logger) -> Self {
+        // Why is this check necessary?
```

Comment:
I guess to avoid have a node self-broadcast? maybe we should allow that?

Reply:
Ah -- I missed the "!"